enumerable.rb

```diff
@@ -0,0 +1,154 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+module Enumerable
+  def my_each()
+    return self unless block_given?
+
+    length.times do |index|
+      yield(self[index])
+    end
+  end
+
+  def my_each_with_index()
+    return self unless block_given?
+
+    length.times do |index|
+      yield(self[index], index)
+    end
+  end
+
+  def my_select()
+    result = []
+
+    return self unless block_given?
+
+    Array(self).my_each do |item|
+      next unless yield(item)
+
+      result.push(item)
+    end
+
+    result
+  end
+
+  def my_all?(argument = nil)
+    Array(self).my_each do |item|
+      if !block_given?
+        if argument.nil? # no argument passed
+          next if item
+        elsif argument.class == Class
+          next if item.is_a? argument
+        elsif (item.is_a? Numeric) || (item.is_a? String) # When it is a value
+          next if item == argument
+        elsif (argument.class == Regexp) && !(item =~ argument).nil?
+          next
+        end
+      elsif yield(item)
+        next
+      end
+      return false
```

Comment:
- [ ] when a Regex is passed as an argument, it returns true if all of the collection matches the Regex